src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -199,6 +203,61 @@ public FormulaValue Eval(string expressionText, RecordValue parameters = null)
             return newValue;
         }
 
+        
+        /// <summary>
+        /// Convert references in an expression to the invariant form
+        /// </summary>
+        /// <param name="expressionText">textual representation of the formula</param>
+        /// <param name="parameters">Type of parameters for formula. The fields in the parameter record can 
+        /// be acecssed as top-level identifiers in the formula. If DisplayNames are used, make sure to have that mapping
+        /// as part of the RecordType
+        /// <returns>The formula, with all identifiers converted to invariant form</returns>
+        public string GetInvariantExpression(string expressionText, RecordType parameters)
+        {
+            return ConvertExpression(expressionText, parameters, toDisplayNames: false);
+        }
+        
+        /// <summary>
+        /// Convert references in an expression to the display form
+        /// </summary>
+        /// <param name="expressionText">textual representation of the formula</param>
+        /// <param name="parameters">Type of parameters for formula. The fields in the parameter record can 
+        /// be acecssed as top-level identifiers in the formula. If DisplayNames are used, make sure to have that mapping
+        /// as part of the RecordType
+        /// <returns>The formula, with all identifiers converted to display form</returns>
+        public string GetDisplayExpression(string expressionText, RecordType parameters)
+        {
+            return ConvertExpression(expressionText, parameters, toDisplayNames: true);
+        }
+
+        private string ConvertExpression(string expressionText, RecordType parameters, bool toDisplayNames)
+        {
+            var formula = new Formula(expressionText);
+            formula.EnsureParsed(TexlParser.Flags.None);
+
+            var extraFunctions = _extraFunctions.Values.ToArray();
+            var resolver = new RecalcEngineResolver(this, parameters, _powerFxConfig.EnumStore.EnumSymbols, extraFunctions);
+            var binding = TexlBinding.Run(
+                new Glue2DocumentBinderGlue(),
+                null,
+                new Core.Entities.QueryOptions.DataSourceToQueryOptionsMap(),
+                formula.ParseTree,
+                resolver,
+                ruleScope: parameters._type,
+                useThisRecordForRuleScope: false,
+                updateDisplayNames: toDisplayNames,
+                forceUpdateDisplayNames: toDisplayNames
+            );
+
```

Comment:
Can this be shared with CheckInternal()?

Reply:
only lines 235-239, which doesn't really give us that much benefit. Might be something to address later if we make the configuration part here more complex (functions, option sets, enums ...)